engine/src/main/java/org/terasology/engine/rendering/opengl/OpenGLMeshBase.java

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.rendering.opengl;
+
+import org.lwjgl.opengl.GL30;
+import org.terasology.engine.rendering.assets.mesh.resouce.VertexResource;
+
+import java.util.List;
+
+public interface OpenGLMeshBase {
+    class VBOContext {
+        private VBOSubBuffer[] entries;
+        private int vbo;
+
+        public static class VBOSubBuffer {
+            int version;
+            int offset;
+            VertexResource resource;
+        }
+    }
+
+    default void updateState(VBOContext state) {
+        GL30.glBindBuffer(GL30.GL_ARRAY_BUFFER, state.vbo);
+        for (int x = 0; x < state.entries.length; x++) {
+            if (state.entries[x].version != state.entries[x].resource.getVersion()) {
+                GL30.glBufferSubData(GL30.GL_ARRAY_BUFFER, state.entries[x].offset, state.entries[x].resource.buffer);
+            }
+        }
+        GL30.glBindBuffer(GL30.GL_ARRAY_BUFFER, 0);
```

Comment:
trying to find a mismatch with the marked vbo so the data is correctly updated during the drawing pass.